File: python/utilidades.py
from typing import Any
def linea(largo: int):
    print("═" * largo) 
    
def encabezado(titulo: str, largo: int):
    linea(largo)
    print(titulo.upper().center(largo))
    linea(largo)
    
def leerNumero(msj:str, fn: Any) -> Any:
      numero: Any = 0
      while True:
          try: 
              numero = fn(input(msj + ': '))
              return numero
          except ValueError:
              print('debe escribir un numero...')
  
def leerTexto(msj: str, aviso: str) -> str:
    texto: str = ''
    while True:
        texto = input(msj + ": ")
        if texto.strip():
            return texto.strip()
        else: 
            print(f'debes escribir el {aviso}... ')
          
                    

File: JS/src/utilidades.js
/* 
funciones de utilidad
 */
import promptSync from "prompt-sync";
const prompt = promptSync();

const center = function(texto='', largo=0){
    let relleno = (largo - texto.length)/2 ;
    let textoCentrado = ' '.repeat(relleno) + texto;
    return textoCentrado;
} 

const linea = function(largo=0) {
    console.log('═'.repeat(largo));
}

const encabezado = function(titulo='', largo=0){
    linea(largo);
    console.log(center(titulo.toUpperCase(),largo));
    linea(largo);
}

const leerNumero = function (msj = '') {
    let numero = 0;
    while (true) {
        numero = Number(prompt(msj + ': '));
        if (!isNaN(numero) && isFinite(numero)) {
            return numero            
        } else {
            console.log("debe escribir un numero...")
        }
    }
}

const leerTexto = function (msj = '', aviso = '') {
    let texto = ''
    while (true) {
        texto = prompt(msj + ": ")
        if (texto.trim()) {
            return texto.trim()
        }else{
            console.log(`debe escribir ${aviso}...`)
        }
    }
}

export {
    encabezado,
    linea,
    leerNumero, 
    leerTexto

}